src/result.rs

```diff
@@ -1,9 +1,74 @@
 use std::convert::From;
-use std::fmt::{Debug, Error};
+use std::fmt::{Debug, Display, Error};
 use std::{fmt, io};
 
 use thiserror::Error;
 
+/// Represents the position, as line and column, within a text format ion document.
+#[derive(Clone, Debug)]
+pub enum TextPosition {
+    /// No text position available; implies binary format is being read.
+    None,
+    /// The position represented as line and column.
+    LineColumn(usize, usize),
+}
+
+#[derive(Clone, Debug)]
+pub struct Position {
+    pub byte_offset: usize,
+    pub line_column: TextPosition,
+}
+
+impl Position {
+    fn with_offset(offset: usize) -> Self {
+        Position {
+            byte_offset: offset,
+            line_column: TextPosition::None,
+        }
+    }
+
+    fn with_line_info(offset: usize, line: usize, column: usize) -> Self {
+        Position {
+            byte_offset: offset,
+            line_column: TextPosition::LineColumn(line, column),
+        }
+    }
+}
+
+impl PartialEq for Position {
```

Comment:
Is there a reason you can't derive `PartialEq` (and `Eq` and `Hash`) for `Position` (and `TextPosition`)?

Reply:
In the latest commit I got rid of the manual implementations and added `fn has_text_position(&self) -> bool` to help out with ease of access.

After thinking more about Zack's comment about converting TextPosition to a struct, I'm kind of leaning towards wanting to eliminate TextPosition entirely and adding line and column info as an `Option<(usize, usize)>` which would then provide the niceties of Option.